compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2130,6 +2130,32 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
       end extension
     end MatchCaseMethods
 
+    type ClassInfo = dotc.core.Types.ClassInfo
+
+    given ClassInfoTypeTest: TypeTest[TypeRepr, ClassInfo] with
+      def unapply(x: TypeRepr): Option[ClassInfo & x.type] = x match
+        case x: (Types.ClassInfo & x.type) => Some(x)
+        case _ => None
+    end ClassInfoTypeTest
+
+    object ClassInfo extends ClassInfoModule:
+      def unapply(x: ClassInfo): (TypeRepr, Symbol, List[TypeRepr], List[Symbol], Option[TypeRepr]) =
+        (x.prefix, x.cls, x.declaredParents, ClassInfoMethods.decls(x), ClassInfoMethods.selfInfo(x))
+    end ClassInfo
+
+    given ClassInfoMethods: ClassInfoMethods with
+      extension (self: ClassInfo)
+        def qualifier: TypeRepr = self.prefix
+        def decls: List[Symbol] = self.decls.toList
+        def declaredParents: List[TypeRepr] = self.declaredParents
```

Comment:
I wouldn't expose `declaredParents` as it's just a less precise version of `parents`: https://github.com/lampepfl/dotty/blob/0dbe9705ec6a319af2365de4e00d3f66154fc133/compiler/src/dotty/tools/dotc/core/Types.scala#L4536

Reply:
But then we would have to always go to the tree which is the thing we are trying to avoid.

Reply:
> But then we would have to always go to the tree

No, we should expose Type#parents which doesn't involve trees.